### PR TITLE
Update to go1.17, and bring up linter, and deps

### PR DESCRIPTION
**Pull request by bnevis-i**
Resolves CVE-2022-41717 and CVE-2022-32149 and general housekeeping

Signed-off-by: Bryon Nevis <bryon.nevis@intel.com>

### CODE REVIEW

**Review by azdagron — APPROVED** [review]
Looks good. I'm going to file an issue to run tests against all supported go versions so we can ensure we keep go1.13 compat.

**Comment by azdagron** [comment]
Oh, already filed https://github.com/spiffe/go-spiffe/issues/206 :)

**Comment by bnevis-i** [comment]
```
Error: ../../../../go/pkg/mod/golang.org/x/net@v0.4.0/http2/transport.go:19:2: package io/fs is not in GOROOT (/opt/hostedtoolcache/go/1.13.15/x64/src/io/fs)
```

Github action isn't using the same Go version that is specified in the Makefile.

File: v2/go.mod [hunk 1]
```diff
@@ -3,11 +3,11 @@ module github.com/spiffe/go-spiffe/v2
 go 1.13
```
Comment (MarcosDY) [inline v2/go.mod:3]: update version to 1.18 here too

**Comment by bnevis-i** [comment]
> Looks good. I'm going to file an issue to run tests against all supported go versions so we can ensure we keep go1.13 compat.

What are the supported go versions?  `io/fs` was introduced in 1.16 and anything less than 1.18 isn't supported anymore.

**Comment by azdagron** [comment]
I think we should move to go1.17. That is the version that most of our dependencies use as a minimum.

File: v2/go.mod (hunk 1 above)
Comment (MarcosDY) [reply v2/go.mod:3]: same for [workflow](https://github.com/spiffe/go-spiffe/blob/main/.github/workflows/pr_build.yaml#L6)

**Comment by bnevis-i** [comment]
what compat mode do you want for the go tidy?

**Comment by bnevis-i** [comment]
Just pushed a version tidied with -go=1.16 AND -go=1.17.  Let me know if you want -compat=1.17

**Comment by azdagron** [comment]
I think running with `-go=1.17` is correct.

**Comment by azdagron** [comment]
We just need to update the .github/workflows/pr_build.yaml to use go1.17 instead of go1.13.

**Comment by bnevis-i** [comment]
> We just need to update the .github/workflows/pr_build.yaml to use go1.17 instead of go1.13.

Amended

```yaml
GO_VERSION: 1.17
```

**Comment by bnevis-i** [comment]
I think this PR needs to be put on hold in order to deal with linter issues.   Lots of issues upgrading the linter :-(

**Comment by azdagron** [comment]
Yikes. Noticed that. Is that something you are planning to work on @bnevis-i? Or should one of us pick it up?

**Comment by bnevis-i** [comment]
> Yikes. Noticed that. Is that something you are planning to work on @bnevis-i? Or should one of us pick it up?

I can give it a shot.  Unfortunately, a good number of the findings require subjective decisions, like what is the min tls version, how much ReadHeaderTimeout is needed.  I can put up a draft PR and get feedback on the right values to use, or I can just wait for someone on the project who knows what the right values are to take care of it.  Which would you like?

**Comment by azdagron** [comment]
I can toss up a a PR real quick

**Comment by azdagron** [comment]
Actually, i'll just push a commit on top of this PR. Then maybe we can rename it something more inclusive of the actual changes we're making.

**Comment by bnevis-i** [comment]
@azdagron Nice job.  I've got to learn to push onto someone else's PR someday.  Neat trick.  It was the right call to resolve the linters yourself.  I would have done every one differently!

**Comment by azdagron** [comment]
> I've got to learn to push onto someone else's PR someday.

It's not terribly hard. You have to add a new local remote to the forked repo, fetch, and then you can just push onto remote PR branch. You can delete the remote afterwards. You of course have to be a codeowner of the repo that is forked, AND it has to be enabled on the PR by the author (which default is controlled by the user? or maybe the org? I don't recall).

**Review by MarcosDY — APPROVED** [review]
LGTM!!!

**Pull request merged**